This patch adds an option to the phone's dropdown menu during a call to view the call history. -skp (www.teckaddict.com)
Index: /usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/activecall-assistant.js
@@ -132,6 +132,10 @@ var ActivecallAssistant = Class.create({
 		var appMenuModel = {
 			visible: true,
             items: [{
+                label: $L('Call History'),
+                command: 'callhistory'
+		  },
+		  {
                 label: $L('Sounds & Ringtones'),
                 command: 'ringtones'
             }]
@@ -143,6 +147,9 @@ var ActivecallAssistant = Class.create({
 	handleCommand: function(event) {
 		if (event.type == Mojo.Event.command) {
 			switch (event.command) {
+				case 'callhistory':
+					this.appAssistant.showCallLog();
+					break;
 				case 'ringtones':
 					MenuController.showRingtones(this.controller.stageController);
 					break;
Index: /usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
+++ /usr/palm/applications/com.palm.app.phone/app/controllers/calllog-assistant.js
@@ -169,7 +169,12 @@ var CalllogAssistant = Class.create({
 			// manually deal with back; always return to dialpad even if there's
 			// no dialpad scene yet created
 			QDLogger.log( "CalllogAssistant#handleCommand", "back!");
-			this.appAssistant.showDialpad();
+			if (this.telListener.callExists() == true) {
+				this.appAssistant.showActiveCall();
+			}
+			else {
+				this.appAssistant.showDialpad();
+			}
 			event.stopPropagation();
 			Event.stop(event);
 		} else if (event.type == Mojo.Event.command) {
@@ -210,7 +215,12 @@ var CalllogAssistant = Class.create({
 	},
     
     switchToDialpad: function(){
-       this.appAssistant.showDialpad();
+	if (this.telListener.callExists() == true) {
+		this.appAssistant.showActiveCall();
+	}
+	else {
+       	this.appAssistant.showDialpad();
+	}
     },
 
 	/* TODO FIXME
